testing/test_pipeline.py

```diff
@@ -189,6 +189,90 @@ def test_pipeline_set_one_hyperparam_level_two_dict():
     assert p["c"].hyperparams == dict()
 
 
+def test_pipeline_update_hyperparam_level_one_flat():
```

Comment:
There might be too many new tests in this file. I feel like just 1 or 2 would be ok.

Reply:
optional fix.

Reply:
won't fix